### DISCUSSION TITLE
[GraphQL API] CreateBranchProtectionRuleInput and UpdateBranchProtectionRuleInput should support merge queue configurations

### DISCUSSION THREAD

**Discussion by zhpeng811** [community/community#77614] [Apps, API and Webhooks] [answered]
### Select Topic Area

Product Feedback

## Background
MergeQueue and its related resource types was added to the GraphQL API on 2023-04-13: https://docs.github.com/en/graphql/overview/changelog#schema-changes-for-2023-04-13-1

## Problem
I'm trying to utilize GraphQL API to create branch protection rules with merge queue configurations, but realized [CreateBranchProtectionRuleInput](https://docs.github.com/en/graphql/reference/input-objects#createbranchprotectionruleinput) and [UpdateBranchProtectionRuleInput](https://docs.github.com/en/graphql/reference/input-objects#updatebranchprotectionruleinput) currently does not support merge queue configurations

## Request
add merge queue configuration support to `CreateBranchProtectionRuleInput` and `UpdateBranchProtectionRuleInput` Input Objects

**Comment by achmadfha** [comment]
you mean like this ?

`input CreateBranchProtectionRuleInput {
  repositoryId: ID!
  pattern: String!
  mergeQueueEnabled: Boolean
  mergeQueueName: String
}

input UpdateBranchProtectionRuleInput {
  branchProtectionRuleId: ID!
  mergeQueueEnabled: Boolean
  mergeQueueName: String
}
`

**Comment by zhpeng811** [comment]
> you mean like this ?
> 
> `input CreateBranchProtectionRuleInput { repositoryId: ID! pattern: String! mergeQueueEnabled: Boolean mergeQueueName: String }
> 
> input UpdateBranchProtectionRuleInput { branchProtectionRuleId: ID! mergeQueueEnabled: Boolean mergeQueueName: String } `

kind of, except merge queue doesn't have a name (it's just attached to the branch protection rules), there are additional configurations that should be also included. See [MergeQueueConfiguration](https://docs.github.com/en/graphql/reference/objects#mergequeueconfiguration) Object and screenshot below: 
![image](https://github.com/community/community/assets/43459469/2ff9c865-ca5a-4534-9b90-dee428f7d0ab)

**Comment by ghostinhershell** [comment]
Hi @zhpeng811, 

Thanks so much for sharing your feedback here in the community. I'll get this filed with the appropriate teams so that this is on their radar. 

Cheers! 🚀

**Reply by tj-smith47** [reply]
Is there somewhere we can track this? It would be very helpful to have some sort of comprehensive guide for common use cases / ways to interact with anything merge queue for either API (I know only GraphQL is available right now) with an example for each confirmed functional feature that people could track / subscribe to for updates. The general community thread seems to have been abandoned, and there are all sorts of mentions of merge queue throughout discussions and documentation, but they are very fragmented, sometimes difficult to piece together - and then once you do, you might find out that what you're actually trying to do isn't fully developed yet / working right, like this. It would be helpful to the community, but also having a de-facto list of "Here is every way merge queues can be interacted with as of today" will likely take some heat off you as well.

**Reply by Niceplace** [reply]
Hello hello, any updates on this ? We will have to do this manually on more than 100 repositories soon and it would have been so much simpler with an API call

**Comment by StephenWithPH** [comment]
FYI- this heavily crosses over with https://github.com/orgs/community/discussions/50893 .

A concrete use case where this would be useful is in https://github.com/integrations/terraform-provider-github/issues/1481

**Reply by chess4ever** [reply]
I would add that it is useful also for other IAC providers like Pulumi.

**Comment by zhpeng811** [comment, marked as answer]
merge queue support have been added for rulesets:https://docs.github.com/en/graphql/reference/enums#repositoryruletype

and since branch protection and ruleset are very similar GitHub probably won't add support to it, marking this discussion as resolved

**Reply by eelco** [reply]
Merge queue enforcing via rulesets is currently broken though
